addons/main/XEH_postInit.sqf

```diff
@@ -35,6 +35,30 @@ if (!GVAR(enable)) exitWith {
 
 GVAR(ammoPenCache) = createHashMap;
 
+["CAManBase", "Local", {
+    params ["_unit", "_isLocal"];
+
+    if (_isLocal && {alive _unit && {GVAR(numWearablePlates) isNotEqualTo 0 && {isNil {((vestContainer _unit) getVariable [QGVAR(plates), nil])}}}}) then {
```

Comment:
Imo it should be the machine that is NOT local to the unit anymore that needs to do something. That is to check if the unit has plates and if it has at least one then sync that to the machine that currently owns the unit. With this plates stay consistent. 

This will make the underlying code a lot smaller.

This event handler fires only on the machines involved in the locality change, that is then at max 2 machines. The one that the unit belonged to and the new owner.

Reply:
on another note. If `isNil` is used, try to only compare a string of a variable name, that is a lot faster in sqf performance wise.

in your case the `isNil` check is not even needed as you can just compare the `QGVAR(plates)` variable with a `isNotEqualTo` and the default return value of `[]` with `getVariable`.

Reply:
I see, how should I refer to the new owner in this context, since `owner _unit` returns 0 on clients?

Reply:
The easiest is to offload that to a CBA event. So you can just do a `CBA_fnc_targetEvent` call and cba will figure it out to which machine to send the data to.